### PR TITLE
Add safe Softmax fusion

### PR DESCRIPTION
**Pull request by robertknight**
ONNX models exported with the latest version of PyTorch apply Where + IsNaN operations to the result of softmax ops that are part of SDPA operations (see https://github.com/pytorch/pytorch/pull/159973). This replicates the "safe softmax" behavior of PyTorch's `scaled_dot_product_attention` operation, which is designed to handle inputs where every element in a row is masked using -inf.

Add an option to the vectorized softmax implementation in rten-vecmath to implement this behavior, a new attribute to the Softmax operation which enables its use and a fusion that combines Softmax + Where + IsNaN operations into a single Softmax with this attribute enabled. Also do the same for the AddSoftmax fusion.

**TODO:**

- [x] Test for attribute in Softmax op
- [x] Test for fusion

### CODE REVIEW

**Comment by robertknight** [comment]
As an aside, I'm wincing slightly at implementing a change to Softmax which means the outputs are not guaranteed to sum to 1 if the "safe" option is enabled, and thus may not be a valid probability distribution. A more principled approach it seems to me would be to treat infinities like `f32::{MIN, MAX}`, then if all inputs are -inf you'll get 1/n for each output (ie. a flat distribution). However, we need to adhere to the behavior defined in the exported ONNX model, which follows PyTorch. The reason PyTorch's safe softmax chose this handling of infinities is explained in https://github.com/pytorch/pytorch/issues/41508.
